### PR TITLE
qa: Verify DBWrapper iterators are taking snapshots

### PR DESCRIPTION
**Pull request by TheBlueMatt**
The LevelDB docs seem to indicate that an iterator will not take
snapshots (even providing instructions on how to do so yourself).
In several of the places we use them, we assume snapshots to have
been taken.

In order to make sure LevelDB doesn't change out from under us
(and to prevent the next person who reads the docs from having the
same fright I did), verify that snapshots are taken in our tests.

### CODE REVIEW

File: src/test/dbwrapper_tests.cpp [hunk 1]
```diff
@@ -204,19 +204,33 @@ BOOST_AUTO_TEST_CASE(iterator_ordering)
     for (int x=0x00; x<256; ++x) {
         uint8_t key = x;
         uint32_t value = x*x;
-        BOOST_CHECK(dbw.Write(key, value));
+        if (!(x & 1))
```
Comment (sipa) [inline src/test/dbwrapper_tests.cpp]: Can I haz braces before indenting?

**Review by sipa — COMMENTED** [review]
utACK

**Comment by mess110** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/11422/commits/517ec7260c8577252decba88bfcf79071a4cd48e

File: src/test/dbwrapper_tests.cpp (hunk 1 above)
Comment (TheBlueMatt) [reply src/test/dbwrapper_tests.cpp]: No, but I fixed it anyway :p.

**Comment by laanwj** [comment]
Yes, iterators not changing under us is an important part of our leveldb usage. I remember in the past this was well-documented. I wonder when they changed the documentation to no longer say this, and what their plans here are?

In any case it's good to check.
utACK bb8376b

**Pull request merged**